aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -1043,23 +1043,35 @@ Tensor reduce_sparse_csr_dim0_cpu_template(const Tensor& sparse, ReductionOp rop
   new_crow_indices[1] = nnz;
 
   Tensor new_values = at::empty({nnz}, values.options());
```

Comment:
It is because, sum.dim_IntList of SparseCsr dispatch will call _sparse_csr_sum.dim_dtype, which output data type will follow input data type.

For the low precision data type (BFloat16, Half, ComplexHalf), it has accuracy issue if directly using low precision as accumulate type. Using Float or ComplexFloat can avoid this problem and it is the same behavior in sum's accumulation. `new_values_acc` is allocated for accumulation -- intermediate variable, `new_values` is the output tensor. They need to be seperated.